### PR TITLE
Fix Grsim Flicker Bug

### LINKED ISSUE #2054: Robots Flicker when using GrSim Vision
### Description of the task
When using GrSim, robots flicker in and out of existence. This is likely a bug in SensorFusion and robot filters and we need to understand why this is a problem with grsim and not our own simulator.

### Acceptance criteria

- [ ] Stop robots from flickering
- [ ] Root cause why this is the case with grsim

### PR DESCRIPTION
**Pull request by LiCody**
### Description
- Made the sensorProto buffer bigger in sensor fusion
- 4 is likely the perfect size because that's how many different frames are sent (Top Yellow, Bottom Yellow, Top Blue, Bottom Blue)

### Testing Done
- Tested with other values to see what size is needed

### Resolved Issues

resolves #2054

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by akhilveeraghanta — COMMENTED** [review]
Awesome for figuring this out so quickly! Does our custom simulator still work for you with this change?

File: src/software/sensor_fusion/threaded_sensor_fusion.h [hunk 1]
```diff
@@ -19,4 +19,5 @@ class ThreadedSensorFusion : public Subject<World>,
     void onValueReceived(SensorProto sensor_msg) override;
 
     SensorFusion sensor_fusion;
+    static constexpr std::size_t DIFFERENT_GRSIM_FRAMES_RECEIVED = 4;
```
Comment (jonathanlew) [inline src/software/sensor_fusion/threaded_sensor_fusion.h]: nit: we usually just use `size_t`

**Review by jonathanlew — COMMENTED** [review]
Looks good, not much to add

**Comment by LiCody** [comment]
From my preliminary testing with the er-force simulator, this change probably won't be needed.

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by amrm3lm — APPROVED** [review]
lgtm

**Pull request merged**